### DISCUSSION TITLE
variable in className not work?

### DISCUSSION THREAD

**Discussion by PeterlitsZo** [tailwindlabs/tailwindcss#7273] [Help]
This is the mini `App.jsx`:

``` javascript
import { useState } from 'react'
import logo from './logo.svg'
import classNames from 'classnames'

function App() {
  const breakPoint = 'md';

  return (
    <div className={classNames(
        "bg-green-300",
        `${breakPoint}:bg-red-300`
    )}>
      Hello
    </div>
  )
}

export default App
```

When I just use `"md:bg-red-300"`, it works. But when I just use `${breakPoint}:bg-red-300`, it does not work. (It means, that when I change the device's width, it do not change the background color). Is it a bug?

---

I am trying to let my mini project online... Oh no, I cannot put it online...

**Comment by PeterlitsZo** [comment]
I know why. Because of `purge`.

**Reply by PeterlitsZo** [reply]
In the version `v3`, there is `content` but not `purge`. I am reading the doc in Chinese version `tailwindcss.cn` but not English version `tailwindcss.com`. That's why I cannot find `purge` in English document.

**Reply by PeterlitsZo** [reply]
Oh my god. How can I use the variable in class?

**Reply by RobinMalfait** [reply]
The biggest issue is that Tailwind is a build-time tool, not a runtime tool. This means that classes created at runtime are not seen by the compiler. We do not execute your code.

Instead, you have a few options:

- Use a safelist where you can a list all the classes that will be generated at runtime (https://tailwindcss.com/docs/content-configuration#safelisting-classes)
- Use a lookup table instead (https://tailwindcss.com/docs/content-configuration#class-detection-in-depth) look at the `Button.jsx` example.

**Reply by PeterlitsZo** [reply]
Thanks for your advice. So the `JIT` means just in build time but not just in runtime...

Well I find that I can use `CSS-in-JS`. And I find [`twin.macro`](https://github.com/ben-rogerson/twin.macro). Will it give me little size? I am not sure.

**Reply by PeterlitsZo** [reply]
I find styled-system. It looks also have shortname (What I want) and also is a CSS-in-JS.

**Reply by JulianKoehler** [reply]
Even after making a production build tailwind won't apply class names saved in variables. I am importing an object which has a property with a string representing a tailwind color class and it won't be applied to the element.

**Comment by mavdotj** [comment]
What I usually do in these cases is create a comment with all the classes that will be used, but this probably isn’t recommended in certain cases.

**Reply by mybearworld** [reply]
If there's no better way to do it ([there usually is](https://tailwindcss.com/docs/content-configuration#dynamic-class-names)), then you should use [`safelist`](https://tailwindcss.com/docs/content-configuration#safelisting-classes) instead.
